stores/s3store/sent.go

```diff
@@ -15,49 +15,28 @@
 package s3store
 
 import (
-	"bytes"
+	"context"
+
+	"github.com/mailchain/mailchain/encoding"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/s3"
-	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/mailchain/mailchain"
-	"github.com/mailchain/mailchain/encoding"
 	"github.com/mailchain/mailchain/internal/mail"
-	"github.com/pkg/errors"
 )
 
 // NewSent creates a new S3 store.
 func NewSent(region, bucket, id, secret string) (*Sent, error) {
-	if region == "" {
-		return nil, errors.Errorf("`region` must be specified")
-	}
-	if bucket == "" {
-		return nil, errors.Errorf("`bucket` must be specified")
-	}
-
-	var creds *credentials.Credentials
-
-	if id != "" && secret != "" {
-		creds = credentials.NewStaticCredentials(id, secret, "")
+	s3Store, err := NewUploader(region, bucket, id, secret)
+	if err != nil {
+		return nil, err
 	}
-	ses := session.Must(session.NewSession(&aws.Config{
-		Region:      aws.String(region),
-		Credentials: creds,
-	}))
 
-	// S3 service client the Upload manager will use.
-	return &Sent{
-		uploader: s3manager.NewUploaderWithClient(s3.New(ses)).Upload, // Create an uploader with S3 client and default options
-		bucket:   bucket,
-	}, nil
+	return &Sent{Uploader: s3Store}, nil
 }
 
 // Sent handles storing messages in S3
 type Sent struct {
-	uploader func(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
-	bucket   string
+	*Uploader
```

Comment:
Try to use composition over inheritance. `uploader = *s3store.Uploader` is preferred rather than`TransactionStore` inheriting from `s3store.Uploader`